drivers/serial/uart_sam0.c

```diff
@@ -1126,8 +1126,9 @@ static const struct uart_driver_api uart_sam0_driver_api = {
 	.configure = uart_sam0_configure,
 	.config_get = uart_sam0_config_get,
 #endif
-	.err_check = uart_sam0_err_check,
+
 #if CONFIG_UART_INTERRUPT_DRIVEN
+	.err_check = uart_sam0_err_check,
```

Comment:
The error I get that prompted that fix is when building with CONFIG_UART_ASYNC_API  but not CONFIG_UART_INTERRUPT_DRIVEN.
Since the actual function uart_sam0_err_check is [guarded by the ifdef](https://github.com/zephyrproject-rtos/zephyr/blob/11df9acd12a86059fea2533a38e4067261ffcecc/drivers/serial/uart_sam0.c#L724) a compile error is raised as the function is not defined. If it must always pre present then it should be moved outside the `#ifdef CONFIG_UART_INTERRUPT_DRIVEN`

Reply:
since .err_check  is a required method, I moved the uart_sam0_err_check block outside of the if guard to allow for compiling when CONFIG_UART_INTERRUPT_DRIVEN is not used. aca350547b8ff15f7f8fbe187ee7e8d48139700b